searx/webapp.py

```diff
@@ -545,6 +545,9 @@ def index_error(output_format, error_message):
 def index():
     """Render index page."""
 
+    # UI
+    advanced_search = request.form.get('advanced_search', request.preferences.get_value('advanced_search'))
```

Comment:
Line 614 (in the search function) should be replaced by the same code.

Reply:
Well, I can commit the change you're suggesting, but the "Invalid settings" banner still appears anyway...